scipy/ndimage/filters.py

```diff
@@ -274,6 +305,9 @@ def gaussian_filter(input, sigma, order=0, output=None,
     truncate : float
         Truncate the filter at this many standard deviations.
         Default is 4.0.
+    weights : ndarray
```

Comment:
You mean a function that does some sort of automatic broadcasting? We'd have to think about what the signature / API would look like, probably better in a subsequent issue/PR.